WcaOnRails/app/models/user.rb

```diff
@@ -357,9 +357,17 @@ def any_kind_of_delegate?
     delegate_status.present?
   end
 
-  def can_edit_users?
-    organizes_comp_with_wca_registration = organized_competitions.not_over.exists?(use_wca_registration: true)
-    admin? || board_member? || results_team? || any_kind_of_delegate? || organizes_comp_with_wca_registration
+  def can_view_all_users?
+    admin? || board_member? || results_team? || any_kind_of_delegate?
+  end
+
+  def can_edit_user?(user)
+    current_user == user || can_view_all_users? || is_organizer_for?(user)
```

Comment:
The first condition should be `self == user`. Note that `current_user` [is just an attribute](https://github.com/thewca/worldcubeassociation.org/blob/19d50e06c4aaf5b40c0915f3f43a50f4d34cf5f9/WcaOnRails/app/models/user.rb#L28), that is [set in a controller](https://github.com/thewca/worldcubeassociation.org/blob/19d50e06c4aaf5b40c0915f3f43a50f4d34cf5f9/WcaOnRails/app/controllers/users_controller.rb#L84) in case we really need it in the model, thus it's neither equivalent to `self` nor always present.